Add orchestrator for supporting multiple processes (#208)
open-telemetry/opentelemetry-go-instrumentation: CHANGELOG.md

## [Unreleased]

### Added

- Add ability to monitor multiple processes. ([#197](https://github.com/open-telemetry/opentelemetry-go-instrumentation/issues/197))

### Changed

> - `pkg/instrumentors` Manager struct method  FilterUnusedInstrumentors is made private.
> - `pkg/opentelemtry/` NewController method now accepts ControllerSettings param.
> - `pkg/process/` Analyzer struct `Close` method is removed.

Reviewer: ```suggestion
- The `FilterUnusedInstrumentors` method of  `"go.opentelemetry.io/auto/pkg/instrumentors".Manager` is removed. ([#208](https://github.com/open-telemetry/opentelemetry-go-instrumentation/pull/208))
- The function signature of `"go.opentelemetry.io/auto/pkg/opentelemtry".NewController` has changed.
  It now accepts `context.Context` and `ControllerSettings` parameters. ([#208](https://github.com/open-telemetry/opentelemetry-go-instrumentation/pull/208))
- The `Close` method of `"go.opentelemetry.io/auto/pkg/process".Analyzer` is removed. ([#208](https://github.com/open-telemetry/opentelemetry-go-instrumentation/pull/208))
```

- The function signature of `"go.opentelemetry.io/auto/offsets-tracker/downloader".DownloadBinary` has changed.

>   It now has an additional flag indicating whether it'll build a dummy app for Go stdlib packages or not. ([#256]<https://github.com/open-telemetry/opentelemetry-go-instrumentation/pull/256>)

Reviewer: ```suggestion
  It now has an additional flag indicating whether it'll build a dummy app for Go stdlib packages or not. ([#256](https://github.com/open-telemetry/opentelemetry-go-instrumentation/pull/256))
```

- The function signature of `"go.opentelemetry.io/auto/offsets-tracker/target".New` has changed.
  It now accepts a flag to determine if the returned `Data` is from the Go stdlib or not. ([#256]<https://github.com/open-telemetry/opentelemetry-go-instrumentation/pull/256>)
- Use UPROBE_RETURN to declare the common uprobe return logic (finding the corresponding context, setting up end time, and sending the event via perf buffer) ([#257]<https://github.com/open-telemetry/opentelemetry-go-instrumentation/pull/257>)
- BASE_SPAN_PROPERTIES as common fields (start time, end time, SpanContext and ParentSpanContext) for all instrumentations events (consistent between C and Go structs). ([#257]<https://github.com/open-telemetry/opentelemetry-go-instrumentation/pull/257>)
- Header guards in eBPF code. ([#257]<https://github.com/open-telemetry/opentelemetry-go-instrumentation/pull/257>)

### Fixed

- Fix context propagation across different goroutines. ([#118](https://github.com/open-telemetry/opentelemetry-go-instrumentation/pull/118))

> - The offset tracker can once again build binaries for the Go stdlib. ([#256]<https://github.com/open-telemetry/opentelemetry-go-instrumentation/pull/256>)

Reviewer: ```suggestion
- The offset tracker can once again build binaries for the Go stdlib. ([#256](https://github.com/open-telemetry/opentelemetry-go-instrumentation/pull/256))
```

## [Unreleased]

> - Add ability to monitor multiple processes. ([#197](https://github.com/open-telemetry/opentelemetry-go-instrumentation/issues/197))

Reviewer: ```suggestion
### Added

- Add ability to monitor multiple processes. ([#197](https://github.com/open-telemetry/opentelemetry-go-instrumentation/issues/197))
```
